app/src/main/java/de/mrkriskrisu/vehicletracking/MainActivity.java

```diff
@@ -23,64 +20,50 @@
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
-        instance = this;
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        if (android.os.Build.VERSION.SDK_INT > 9) {
-            StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
-            StrictMode.setThreadPolicy(policy);
-        }
+        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
+        StrictMode.setThreadPolicy(policy);
         createNotificationChannel();
 
-        inpBahnID = MainActivity.getInstance().findViewById(R.id.inpBahnID);
+        inpBahnID = this.findViewById(R.id.inpBahnID);
 
         buttonScan = findViewById(R.id.btnCapture);
-        buttonScan.setOnClickListener(new CaptureManager());
+        buttonScan.setOnClickListener(new CaptureManager(this));
 
         buttonLocate = findViewById(R.id.btnLocate);
-        buttonLocate.setOnClickListener(new LocateManager());
+        buttonLocate.setOnClickListener(new LocateManager(this));
 
         wifiManager = (WifiManager) getApplicationContext().getSystemService(Context.WIFI_SERVICE);
 
         if (!wifiManager.isWifiEnabled())
             wifiManager.setWifiEnabled(true);
 
-        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_WIFI_STATE) != PackageManager.PERMISSION_GRANTED)
-            showPermissionPopup();
-        else if (ContextCompat.checkSelfPermission(this, Manifest.permission.CHANGE_WIFI_STATE) != PackageManager.PERMISSION_GRANTED)
-            showPermissionPopup();
-        else if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_NETWORK_STATE) != PackageManager.PERMISSION_GRANTED)
-            showPermissionPopup();
-        else if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED)
-            showPermissionPopup();
-        else if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED)
-            showPermissionPopup();
-        else if (ContextCompat.checkSelfPermission(this, Manifest.permission.INTERNET) != PackageManager.PERMISSION_GRANTED)
-            showPermissionPopup();
+        //check that we have all necessary permission, if not prompt the user
+        String[] permissions = new String[] {
+                Manifest.permission.ACCESS_WIFI_STATE,
+                Manifest.permission.CHANGE_WIFI_STATE,
+                Manifest.permission.ACCESS_NETWORK_STATE,
+                Manifest.permission.ACCESS_COARSE_LOCATION,
+                Manifest.permission.ACCESS_FINE_LOCATION,
+                Manifest.permission.INTERNET
+        };
+        for (String permission : permissions) {
+            if (ContextCompat.checkSelfPermission(this, permission) != PackageManager.PERMISSION_GRANTED)
```

Comment:
I am not entirely satisfied with this solution, because the popup can be opened several times in the worst case. Can you put a break in there when the popup is confirmed?